### PR TITLE
story/WOR-48/refactor-droits-et-erreurs

### PR DESCRIPTION
**Pull request by remirubis**
## Description

he changes involve moving and editing the logic of the api.response file, removing __v from the exhibitor model, editing routes to add middlewares, implementing access token for tests, updating speaker tests with new API response, and updating controllers for tickets and users to use the new API response class.

## Changes

- Remove `__v` in json in `exhibitor.model.ts`
- Move `api.response.ts` file from `response` folder to `utils`
- Add 401 and 403 in `error.mock.ts`
- Edit middlewares to implement new api response
- Add new function `omitTimestamp` who reduce line width in exhibitors tests
- Edit `ticket` and `user` controllers to fit with new API response class
- Edit `exhibitors` tests to support access token
- Add in `exhibitors` tests test cases for user status
- Edit `tickets` tests to support new api response
- Edit `users` tests to support new api reponse

## To be done in another ticket

- Add mocks for `users` tests
- Add mocks for `tickets` tests

### CODE REVIEW

File: src/__tests__/mockups/error.mock.ts [hunk 1]
```diff
@@ -5,6 +5,18 @@ export const error = {
...
+  401: {
+    name: 'Error',
+    httpStatusCode: 401,
+    description: 'Must be authenticated.',
+    isOperational: true,
+  },
+  403: {
+    name: 'Error',
+    httpStatusCode: 403,
+    description: "You don't have the right access to execute this request.",
```
Comment (iti98) [inline src/__tests__/mockups/error.mock.ts:17]: Wording : "the access right" instead of "the right access"

**Review by iti98 — APPROVED** [review]

**Review by BrystoQ — APPROVED** [review]
LGTM

File: src/__tests__/user.test.ts [hunk 2]
```diff
@@ -32,50 +34,62 @@ describe('Users routes', () => {
...
       expect(response.status).toBe(200);
-      expect(response.body).toHaveProperty('id');
-      expect(response.body).toHaveProperty('firstname');
-      expect(response.body).toHaveProperty('lastname');
-      expect(response.body).toHaveProperty('email');
-      expect(response.body).not.toHaveProperty('password');
-      expect(response.body).toHaveProperty('role');
-      expect(response.body.role).toBe(1930);
+      expect(response.body.data).not.toHaveProperty('password');
+      expect(response.body.data.role).toBe(1930);
```
Comment (Maengdok) [inline src/__tests__/user.test.ts:74]: Un mock up aurait été plus appréciable

File: src/__tests__/speaker.test.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ describe('Speakers routes', () => {
       });
 
       expect(response.status).toBe(200);
-      expect(response.body.role).toBe(Roles.SPEAKER);
+      expect(response.body.data.role).toBe(Roles.SPEAKER);
     });
 
     it('should return 403 if common user try to create a user with speaker role', async () => {
```
Comment (Maengdok) [inline src/__tests__/speaker.test.ts:45]: Pas de tests de valeurs mais uniquement des tests de Status Code générés par la réponse ?

**Review by Maengdok — APPROVED** [review]
Kind of OK for me

File: src/__tests__/mockups/error.mock.ts (hunk 1 above)
Comment (Maengdok) [reply src/__tests__/mockups/error.mock.ts:17]: You don't have the right access to execute this request. : Tu n'as pas le bon accès pour exécuter cette requête.
You don't have the access right to execute this request.:  Tu n'as pas le droit d'accès pour exécuter cette requête.

Rien de mauvais dans les deux phrases.

Comment (iti98) [reply src/__tests__/mockups/error.mock.ts:17]: Ok sorry about it

File: src/__tests__/user.test.ts (hunk 2 above)
Comment (remirubis) [reply src/__tests__/user.test.ts:74]: Bah il sera appreciable quand tu le feras connard.

File: src/__tests__/speaker.test.ts (hunk 3 above)
Comment (remirubis) [reply src/__tests__/speaker.test.ts:45]: Merci de te proposer, je créé le ticket et je te l'assigne.

Ton chef de projet (quand ça m'arrange)

**Pull request merged**